modules/tinymce/src/plugins/fullscreen/test/ts/browser/FullScreenPluginTest.ts

```diff
@@ -33,78 +47,108 @@ UnitTest.asynctest('browser.tinymce.plugins.fullscreen.FullScreenPluginTest', (s
 
   const lastEventArgs = Cell(null);
 
-  const cAssertEditorAndLastEvent = (label: string, state: boolean) =>
-    Chain.control(
+  const cAssertApiAndLastEvent = (label: string, state: boolean) =>
+    Chain.label(
+      `${label}: fullscreen API and event state should return ${state}`,
       Chain.fromChains([
         Chain.op((editor) => Assertions.assertEq('Editor isFullscreen', state, editor.plugins.fullscreen.isFullscreen())),
         Chain.op(() => Assertions.assertEq('FullscreenStateChanged event', state, lastEventArgs.get().state))
-      ]),
-      Guard.addLogging(label)
+      ])
     );
 
-  const cAssertFullscreenClass = (label: string, shouldExist: boolean) => {
+  const cAssertHtmlAndBodyState = (label: string, shouldExist: boolean) => {
     const selector = shouldExist ? 'root:.tox-fullscreen' : 'root::not(.tox-fullscreen)';
-    const label2 = `Body and Html should ${shouldExist ? '' : 'not '}have "tox-fullscreen" class`;
-    return Chain.control(
-      Chain.fromChains([
+    return Chain.label(
+      `${label}: Body and Html should ${shouldExist ? '' : 'not'} have "tox-fullscreen" class`,
+      Chain.fromIsolatedChains([
         Chain.inject(SugarBody.body()),
         UiFinder.cFindIn(selector),
         Chain.inject(SugarElement.fromDom(document.documentElement)),
         UiFinder.cFindIn(selector)
-      ]),
-      Guard.addLogging(`${label}: ${label2}`)
+      ])
     );
   };
 
-  const cCloseOnlyWindow = Chain.control(
-    Chain.op((editor: Editor) => {
-      const dialogs = () => UiFinder.findAllIn(SugarBody.body(), '[role="dialog"]');
-      Assertions.assertEq('One window exists', 1, dialogs().length);
-      editor.windowManager.close();
-      Assertions.assertEq('No windows exist', 0, dialogs().length);
-    }),
-    Guard.addLogging('Close window')
-  );
+  const cAsssertEditorContainerAndSinkState = (label: string, shouldExist: boolean) =>
+    Chain.label(
+      `${label}: Editor container and sink should ${shouldExist ? '' : 'not'} have "tox-fullscreen" class and z-index`,
+      Chain.fromChains([
+        Chain.fromIsolatedChains([
```

Comment:
It _is_ 7PM on a Friday, but I'm finding this rather hard to follow. I'm wondering if NamedChains - much as I dislike using them - might be more readable?